### PR TITLE
Create Faces 4.0 CDI Test for 24631

### PR DESCRIPTION
**Pull request by volosied**
for https://github.com/OpenLiberty/open-liberty/issues/24679  (add tests issue) 
for #24631 ( original release bug issue) 

depends on https://github.com/OpenLiberty/open-liberty/pull/24636 & #24632

### CODE REVIEW

File: dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/faces40/Faces40CDISessionPersistence.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
+package com.ibm.ws.cdi.jee.faces40;
```
Comment (volosied) [inline dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/faces40/Faces40CDISessionPersistence.java:10]: Copyright

File: dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/faces40/app/WEB-INF/web.xml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        <url-pattern>*.xhtml</url-pattern>
+    </servlet-mapping>
+
+        
+    <context-param>
+        <param-name>jakarta.faces.CLIENT_WINDOW_MODE</param-name>
+        <param-value>url</param-value>
+    </context-param>
+    
+</web-app>
```
Comment (volosied) [inline dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/faces40/app/WEB-INF/web.xml]: fix newlines

File: dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/faces40/Faces40CDISessionPersistence.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    }
+
+    @Test
+    public void testFaceletCDISessionPersistenceViaRestart() throws Exception {
+
+        URL defaultURL = new URL("http://" + server.getHostname() + ":" + server.getHttpDefaultPort() + "/" + APP_NAME);;
+
+        try (WebClient webClient = new WebClient()) {
+
+            // webClient.getOptions().setThrowExceptionOnFailingStatusCode(false);
```
Comment (benjamin-confino) [inline dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/faces40/Faces40CDISessionPersistence.java]: We can remove this commented line.

File: dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/faces40/Faces40CDISessionPersistence.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+            HtmlInput input = (HtmlInput) page.getElementById("form:data");
+            input.setValueAttribute("indexdata");
+
+            button = (HtmlElement) page.getElementById("form:submitButton");
+            page = button.click();
+
+            LOG.info(page.asXml());
+
+            LOG.info("Stopping server at " + System.nanoTime());
+            server.stopServer(true, false); // (postStopServerArchive, boolean forceStop)
```
Comment (benjamin-confino) [inline dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/faces40/Faces40CDISessionPersistence.java:114]: We can remove this comment

File: dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/faces40/app/simpleFlow/simpleFlow-2.xhtml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ <html xmlns="http://www.w3.org/1999/xhtml" 
+    xmlns:h="jakarta.faces.html">
+<h:head>
+    <title>Page2</title>
+</h:head>
+<h:body>
+    <h:form id="form">  
+        Page 2
+    <h:inputText id="data" value="#{dataBean.data}" /><br/>  
+    <h:commandButton id="refresh" value="Submit" />
```
Comment (volosied) [inline dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/faces40/app/simpleFlow/simpleFlow-2.xhtml]: change value to refresh

**Comment by volosied** [comment]
#build

File: dev/com.ibm.ws.cdi.jee_fat/publish/servers/cdi40SessionPersistenceServer/server.xml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+<server description="Server for testing Session Persistence with Faces/CDI ">
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.cdi.jee_fat/publish/servers/cdi40SessionPersistenceServer/server.xml:1]: Missing copyright here

File: dev/com.ibm.ws.cdi.jee_fat/publish/servers/cdi40SessionPersistenceServer/bootstrap.properties [hunk 7]
```diff
@@ -0,0 +1,2 @@
+bootstrap.include=../testports.properties
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.cdi.jee_fat/publish/servers/cdi40SessionPersistenceServer/bootstrap.properties:1]: Missing copyright here

File: dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/jsf/simple/SimpleJsfBean.java [hunk 8]
```diff
@@ -1,14 +1,11 @@
 /*******************************************************************************
- * Copyright (c) 2017 IBM Corporation and others.
+ * Copyright (c) 2023 IBM Corporation and others.
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/jsf/simple/SimpleJsfBean.java]: Why the updated copyright here?

File: dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/faces40/app/index.xhtml [hunk 9]
```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
+<!--
+    Copyright (c) 2022 IBM Corporation and others.
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/faces40/app/index.xhtml]: Should this be 2023?

File: dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/FATSuite.java [hunk 10]
```diff
@@ -17,6 +17,7 @@
 import org.junit.runners.Suite.SuiteClasses;
 
 import com.ibm.ws.cdi.jee.ejbWithJsp.JEEInjectionTargetTest;
+import com.ibm.ws.cdi.jee.faces40.Faces40CDISessionPersistence;
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/FATSuite.java:20]: Should we update the copyrights for this change?

File: dev/com.ibm.ws.cdi.jee_fat/build.gradle [hunk 11]
```diff
@@ -10,7 +10,39 @@
  * Contributors:
  *     IBM Corporation - initial API and implementation
  *******************************************************************************/
+
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.cdi.jee_fat/build.gradle:13]: Think this might require a copyright update

File: dev/com.ibm.ws.cdi.jee_fat/bnd.bnd [hunk 12]
```diff
@@ -81,4 +81,11 @@ tested.features: \
 	com.ibm.websphere.javaee.el.3.0;version=latest,\
 	com.ibm.websphere.javaee.jaxrs.2.0;version=latest,\
 	com.ibm.websphere.javaee.interceptor.1.2;version=latest,\
-	com.ibm.websphere.javaee.annotation.1.1;version=latest
+	com.ibm.websphere.javaee.annotation.1.1;version=latest,\
+        net.sourceforge.htmlunit:htmlunit;version=2.44.0,\
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.cdi.jee_fat/bnd.bnd:85]: Does this require a copyright update?

File: dev/com.ibm.ws.cdi.jee_fat/publish/servers/cdi40SessionPersistenceServer/bootstrap.properties (hunk 7 above)
Comment (volosied) [reply dev/com.ibm.ws.cdi.jee_fat/publish/servers/cdi40SessionPersistenceServer/bootstrap.properties:1]: The FAT here doesn't have copyrights here, so I'm excluding it too.

File: dev/com.ibm.ws.cdi.jee_fat/publish/servers/cdi40SessionPersistenceServer/server.xml (hunk 6 above)
Comment (volosied) [reply dev/com.ibm.ws.cdi.jee_fat/publish/servers/cdi40SessionPersistenceServer/server.xml:1]: The FAT here doesn't have copyrights here, so I'm excluding it too.

**Comment by volosied** [comment]
#build

File: dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/jsf/simple/SimpleJsfBean.java (hunk 8 above)
Comment (volosied) [reply dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/jsf/simple/SimpleJsfBean.java]: oops

**Comment by volosied** [comment]
#build

**Review by isaacrivriv — DISMISSED** [review]

File: dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/faces40/app/index.xhtml [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+    are made available under the terms of the Eclipse Public License 2.0
+    which accompanies this distribution, and is available at
+    http://www.eclipse.org/legal/epl-2.0/
+    
+    SPDX-License-Identifier: EPL-2.0
+    Contributors:
+        IBM Corporation - initial API and implementation
+ -->
+<html xmlns="http://www.w3.org/1999/xhtml"
+    xmlns:h="http://xmlns.jcp.org/jsf/html">
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.cdi.jee_fat/fat/src/com/ibm/ws/cdi/jee/faces40/app/index.xhtml:14]: May want to update this to the new namespace jakarta.faces.html

**Comment by volosied** [comment]
Stuck in `Assembling Open Liberty code` for 10 hours.... :/

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
#libby

File: dev/com.ibm.ws.cdi.jee_fat/publish/servers/cdi40SessionPersistenceServer/server.xml [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+    <javaPermission codebase="${shared.resource.dir}/derby/derby.jar" className="javax.management.MBeanServerPermission" name="createMBeanServer"/>
+    <javaPermission codebase="${shared.resource.dir}/derby/derby.jar" className="javax.management.MBeanPermission" name="*" actions="registerMBean,unregisterMBean"/>
+    <javaPermission codebase="${shared.resource.dir}/derby/derby.jar" className="javax.management.MBeanTrustPermission" name="register"/>
+    <javaPermission codebase="${shared.resource.dir}/derby/derby.jar" className="java.lang.RuntimePermission" name="createClassLoader"/>
+    <javaPermission codebase="${shared.resource.dir}/derby/derby.jar" className="java.lang.RuntimePermission" name="getClassLoader"/>
+    <javaPermission codebase="${shared.resource.dir}/derby/derby.jar" className="java.lang.RuntimePermission" name="setContextClassLoader"/>
+    <javaPermission codebase="${shared.resource.dir}/derby/derby.jar" className="java.lang.RuntimePermission" name="getProtectionDomain"/>
+    <javaPermission codebase="${shared.resource.dir}/derby/derby.jar" className="java.sql.SQLPermission" name="deregisterDriver"/>
```
Comment (volosied) [inline dev/com.ibm.ws.cdi.jee_fat/publish/servers/cdi40SessionPersistenceServer/server.xml:32]: Permissions were copied from other server.xmls: https://github.com/OpenLiberty/open-liberty/blob/2fd4a880754c37a988c5ed9ac4f1ea5988e465d6/dev/com.ibm.ws.ejbcontainer.timer.persistent_fat/publish/servers/com.ibm.ws.ejbcontainer.timer.persistent.fat.PersistentTimerRestartServer/server.xml#L35

**Comment by volosied** [comment]
@benjamin-confino  I see there's a conflict (which I'll fix), but can we still merge this into the the CDI fat?

**Comment by volosied** [comment]
#build

**Review by Azquelt — APPROVED** [review]

**Comment by Azquelt** [comment]
#libby

**Pull request merged**